src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): object {
+  return items.reduce((
+    groupOfObejcts: GroupsMap<T>,
+    singleObject,
+  ) => {
+    const groupKey = singleObject[key] as string;
+
+    if (!Object.prototype.hasOwnProperty.call(groupOfObejcts, groupKey)) {
+      Object.assign(groupOfObejcts, { [groupKey]: [] });
+    }
```

Comment:
Use advanced techniques:
```suggestion
    groupOfObejcts[groupKey] ??= [];
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment